src/scmap.rs

```diff
@@ -0,0 +1,69 @@
+#![allow(clippy::result_unit_err)]
+#![allow(clippy::missing_errors_doc)]
+
+use crate::{ScMap, ScMapEntry, ScVal, Validate};
+extern crate alloc;
+use alloc::vec::Vec;
+
+impl ScMap {
+    pub fn sorted_from_entries<I: Iterator<Item = ScMapEntry>>(entries: I) -> Result<ScMap, ()> {
+        let mut v: Vec<ScMapEntry> = entries.collect();
+        // TODO: Add tests that prove order consistency of ScVal with RawVal. https://github.com/stellar/rs-stellar-xdr/issues/117
+        v.sort_by(|a, b| a.key.cmp(&b.key));
+        let m = ScMap(v.try_into()?);
+        // `validate` here will further check that there are no duplicates.
+        m.validate()?;
+        Ok(m)
+    }
+
+    pub fn sorted_from_pairs<K, V, I: Iterator<Item = (K, V)>>(pairs: I) -> Result<ScMap, ()>
+    where
+        ScVal: From<K> + From<V>,
```

Comment:
It might create more flexibility to make this use the Into conversion, or better yet, the TryInto conversion. Using Into instead of From will make it work for types that implement only the Into conversion. Using the Try version will make it work with types that don't offer an errorless conversion, like Vec.

Reply:
Done.